lib/portage/package/ebuild/_ipc/QueryCommand.py

```diff
@@ -149,9 +149,10 @@ def _elog(self, elog_funcname, lines):
         elog_func = getattr(elog_messages, elog_funcname)
         global_havecolor = portage.output.havecolor
         try:
-            portage.output.havecolor = self.settings.get(
-                "NOCOLOR", "false"
-            ).lower() in ("no", "false")
+            portage.output.havecolor = not no_color(
```

Comment:
Can we have a function which takes self.settings as input and handles all of this? I don't want to have to keep repeating the same logic in a few places, and it's hard to read because of the double negative.